opm/simulators/flow/BlackoilModelEbos.hpp

```diff
@@ -677,16 +677,47 @@ namespace Opm {
             return pvSumLocal;
         }
 
+        double computeCnvErrorPv(const std::vector<Scalar>& B_avg, double dt)
+        {
+            double errorPV{};
+            const auto& ebosModel = ebosSimulator_.model();
+            const auto& ebosProblem = ebosSimulator_.problem();
+            const auto& ebosResid = ebosSimulator_.model().linearizer().residual();
+            const auto& gridView = ebosSimulator().gridView();
+            ElementContext elemCtx(ebosSimulator_);
+
+            for (const auto& elem: elements(gridView, Dune::Partitions::interiorBorder))
+            {
+                elemCtx.updatePrimaryStencil(elem);
+                elemCtx.updatePrimaryIntensiveQuantities(/*timeIdx=*/0);
+                const unsigned cell_idx = elemCtx.globalSpaceIndex(/*spaceIdx=*/0, /*timeIdx=*/0);
+                const double pvValue = ebosProblem.referencePorosity(cell_idx, /*timeIdx=*/0) * ebosModel.dofTotalVolume( cell_idx );
```

Comment:
Does `dofTotalVolume` include modifiers such as the net-to-gross factor?  Could it ever be zero?

Reply:
The dofTotalVolume does not include modifiers but the porosity does if I remember correctly. (Doesn't porv from eclipseState include all static modifications to the porosity)  

https://github.com/OPM/opm-simulators/blob/f2c1496c6e8a4ee813080aedf649bb4ab94c6266/ebos/eclproblem.hh#L2372

Reply:
> The dofTotalVolume does not include modifiers but the porosity does if I remember correctly.

Ah, right.

> Doesn't porv from eclipseState include all static modifications to the porosity

You're right.  If we're using this definition of porosity (i.e., `PORV / volume`)&mdash;as opposed to "just" the `PORO` data from an input file&mdash;then we implicitly account for net-to-gross and other effects in the porosity and everything is consistent.